packages/media-utils/src/utils/validate-mime-type.ts

```diff
@@ -11,10 +11,15 @@ import { UploadError } from './upload-error';
 /**
  * Verifies if the caller (e.g. a block) supports this mime type.
  *
- * @param file         File object.
- * @param allowedTypes List of allowed mime types.
+ * @param file                        File object.
+ * @param allowedTypes                List of allowed mime types.
+ * @param [typesNotSupportedByServer] List of types not supported by the server.
  */
-export function validateMimeType( file: File, allowedTypes?: string[] ) {
+export function validateMimeType(
```

Comment:
I think `validateMimeTypeForUser()` is actually the better place for this check. That function already compares the file with what's supported by WordPress.

But here, `allowedTypes` refers to the mime types supported/expected by a block, which can also be omitted (thus the early return in this function).

Reply:
Ok, I put it in `validateMimeTypeForUser` at first and wasn't sure if that was right. Do you think this warrents a new check call "mimeTypesForServer" or something like that?

Reply:
That could be an option, but then I would not export it from the package, to keep it private. This way we don't have to change the function signature of the exported `validateMimeTypeForUser` function.

Reply:
I switched to a separate validation helper for the uploader in the latest version and added a test for the helper. Still need to test to validate all the recent changes.